service/src/main/java/bio/terra/profile/service/profile/flight/update/UpdateProfileRecordStep.java

```diff
@@ -0,0 +1,38 @@
+package bio.terra.profile.service.profile.flight.update;
+
+import bio.terra.profile.db.ProfileDao;
+import bio.terra.profile.service.profile.exception.ProfileNotFoundException;
+import bio.terra.profile.service.profile.model.BillingProfile;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.StepStatus;
+import javax.annotation.Nullable;
+
+public record UpdateProfileRecordStep(
+    ProfileDao profileDao,
+    BillingProfile profile,
+    @Nullable String description,
+    @Nullable String billingAccountId)
+    implements Step {
+
+  @Override
+  public StepResult doStep(FlightContext flightContext) {
+    if (profileDao.updateProfile(profile.id(), description, billingAccountId)) {
+      return StepResult.getStepResultSuccess();
+    } else {
+      return new StepResult(
+          StepStatus.STEP_RESULT_FAILURE_FATAL,
```

Comment:
Could this fail for something non-fatal, like a temporary database connection issue?

Reply:
I expect that would result in `updateProfile` throwing an uncaught (😢) exception. `updateProfile` will only return false if no profile was found, in which case we should bail out of the flight IMO